src/components/screens/FormScreen.tsx

```diff
@@ -3,36 +3,115 @@ import styles from "./FormScreen.module.css";
+import { fromZodError } from "zod-validation-error";
 
 function FormScreen() {
-  const [isBeingTurnedOff, setisBeingTurnedOff] = useState(false)
+  const formMethods = useForm<PaniniFormData>();
+  const { handleSubmit } = formMethods;
+  const navigate = useNavigate();
+  const onSubmit: SubmitHandler<PaniniFormData> = async (data) => {
+    try {
+      const parsedData = paniniSchema.parse(formatData(data));
+      console.log(parsedData);
+      const serverResponse = await postPayload(parsedData);
+      console.log(serverResponse);
+      setisBeingTurnedOff(true);
+      setTimeout(() => {
+        navigate("/success", {state: {imageUrl: serverResponse.imageUrl}});
+      }, 1000);
+    } catch (error) {
+      alert(fromZodError(error)); //using a library to make the error message readable to user
+    }
+  };
 
-  const handleOrder = () => {
-    setisBeingTurnedOff(true)
-  }
+  const [isBeingTurnedOff, setisBeingTurnedOff] = useState(false);
 
-  const topFormWrapper = classNames(styles.wholeFormContainer, isBeingTurnedOff ? styles.turnOffForm : "")
+  const topFormWrapper = classNames(
+    styles.wholeFormContainer,
+    isBeingTurnedOff ? styles.turnOffForm : ""
+  );
 
   return (
-    <div className={topFormWrapper}>
-      <div className={styles.header}>
-        Panini Creator
-        <button className={styles.randomizeButton}>
-          <Dices className={styles.dices} />
-          randomize panini
-        </button>
-      </div>
-      <div className={styles.formElementsContainer}>
-        <FormModuleBase name="Configure Base" />
-        <FormModuleExtras name="Configure Extras" />
-        <FormModuleFinal name="Finalize Order" onOrder={handleOrder} />
+    <FormProvider {...formMethods}>
+      <div className={topFormWrapper}>
+        <div className={styles.header}>
+          Panini Creator
+          <button className={styles.randomizeButton}>
+            <Dices className={styles.dices} />
+            randomize panini
+          </button>
+        </div>
+        <div className={styles.formElementsContainer}>
+          <FormModuleBase name="Configure Base" />
+          <FormModuleExtras name="Configure Extras" />
+          <FormModuleFinal
+            name="Finalize Order"
+            onOrder={handleSubmit(onSubmit)}
+          />
+        </div>
       </div>
-    </div>
+    </FormProvider>
   );
 }
 
 export default FormScreen;
+
+function formatData(data: PaniniFormData) {
+  const returnObject: SandwichPayload = {
+    sandwichName: data.sandwichName.toString(),
+    cutlery: data.cutlery[0] ? true : false,
+    napkins: data.napkins[0] ? true : false,
+    base: {
+      bread: data.bread.toString(),
+      cheese: data.cheese,
+      meat: data.meat,
+      dressing: data.dressing,
+      vegetables:
+        data.vegetables[0].length > 0 ? data.vegetables[0].split(",") : [],
+    },
+    extras: {
+      egg: data.egg,
+      spreads: data.spreads[0].length > 0 ? data.spreads[0].split(",") : [],
```

Comment:
**nitpick:** Let's move these logic outside component file to another, more related file, eg: `src/api/formatters.ts` (formatter), and `src/api/client.ts` (postRequest)